### DISCUSSION TITLE
Do Sonoff devices that have temp probes like the THR320D have pull-up resistors on the input?

### DISCUSSION THREAD

**Discussion by wayner9** [arendst/Tasmota#19996] [Q&A] [answered]
I tried building a homemade cable for a THR320D.  I put two sensors and it was working fine.  But it stopped working after a few hours.  One possible reason is that you need a 4.7k pull up resistor.  I am trying again, so far, so good, but it has only been a short time.  The temp probe that I bought from Sonoff works fine and doesn't die after a while.

I assumed that there was an internal resistor in line between the data and power lines.  But perhaps not and the Sonoff temp probe has the resistor in the probe.

**Comment by sfromis** [comment, marked as answer]
My guess would be no pullup in the probe, but maybe a rather weak pullup in the device, not designed for more than one sensor. As I wrote in the other thread:
> In case of trouble, I'd try adding extra pullup when using multiple DS18B20 sensors. IDK how much the THR320D already has.

More generally, getting cheap DS18B20 probes can be quite a lottery.
https://github.com/cpetrich/counterfeit_DS18B20

**Reply by wayner9** [reply]
Thanks, it just wasn't clear to me how the "stock" Sonoff probe was implementing a pullup resistor, if at all.  That's why I tried without at first.  If I had some spare RJ9 cables I could try measuring resistance across the pins to see if it is on the probe or the THR320D unit.

That's a pretty thorough examination of potentially fake DS18B20 probes.  Strange that they didn't include Amazon as a spot to purchase from.

I do have cheap DS18B20 probes as I recently bought a ten pack of 5m long cables - either on Amazon or AliExpress.

But I also bought some in 2014 that are still in use and working fine.  I think they were bought on Adafruit which may make it more likely that they are legit.

**Reply by sfromis** [reply]
Amazon or Aliexpress is basically the same sellers from China, just as bad, without much focus on sourcing or quality. Not always bad, just not very predictable.

**Comment by lalo-uy** [comment]
Looking at the serial numbers a few of sonoff probes for TH10 showed not
original sensors.
Its a matter of luck.
